packages/open-payments/src/client/incoming-payment.ts

```diff
@@ -151,3 +188,19 @@ export const validateCreatedIncomingPayment = (
 
   return validateIncomingPayment(payment)
 }
+
+export const validateCompletedIncomingPayment = (
+  payment: IncomingPayment
+): IncomingPayment => {
+  const { completed, expiresAt } = payment
+
+  if (new Date(expiresAt).getTime() <= Date.now()) {
```

Comment:
```suggestion
  if (expiresAt && new Date(expiresAt).getTime() <= Date.now()) {
```
Since it's possible `expiresAt` can be undefined in Open Payments. Otherwise, if its undefined, it'll be an `Invalid Date`. 

I'm going to enable strict type checking in `open-payments` so TypeScript complains about this next time

Reply:
I would argue this seems to be like a check mostly necessary at the server receiving this request (which we don't explicitly do based on `expiresAt`, we use the implicit `state`: https://github.com/interledger/rafiki/blob/33f146f0c793bc8915e50c2b3f785d2ae6020742/packages/backend/src/open_payments/payment/incoming/service.ts#L286-L309).

This check could throw even if the non-expired incoming payment was completed successfully, when the `expiryDate` was right at the time of the request